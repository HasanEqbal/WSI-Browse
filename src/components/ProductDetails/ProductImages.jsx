import React, { useState } from 'react';
import './ProductImages.scss';
import Overlay from '../Overlay/Overlay'

export default function ProductsImages({ product }) {
    const [mainImage, setMainImage] = useState(product[0].hero)
    const [displayCarosuel, setDisplayCarosuel] = useState(false)
    let { images } = product[0]
    return (
        <React.Fragment>
            {<div className="medium-6 columns">
                <a>
                    <img className="thumbnail" onClick={() => { setDisplayCarosuel(true) }} src={mainImage} />
                </a>
                <div className="row small-up-8">
                    {images ? images.map((image, index) => {
                        return (<div key={index} className="column">
                            <a>
                                <img className="thumbnail" src={image.href} onClick={(e) => { setMainImage(e.target.src) }} />
                            </a>
                        </div>)
                    }) : <h4>Product Does not have alternative images</h4>}
                </div>
            </div>}
            {displayCarosuel && <Overlay images={images} />}
        </React.Fragment >
    )
}


