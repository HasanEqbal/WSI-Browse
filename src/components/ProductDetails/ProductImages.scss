.small-up-8 {
  margin-top: 10px
}
p{
  font-size: 14px;
}
.small-up-8 .column{
  padding-left: 0px
}
