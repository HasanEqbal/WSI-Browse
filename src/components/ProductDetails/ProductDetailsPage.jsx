import React from 'react';
import ProductDetails from './ProductDetails';
import ProductImages from './ProductImages';
import SimilarProductRenderer from '../SimilarProducts/SimilarProductsRenderer';
import loadingIndicator from '../../Images/loading-indicator-main.jpeg';

export default function ProductDetailsPage({ productsList, match }) {
    return (
        <React.Fragment>
            {productsList.groups ? productsList.groups.filter(item => item.id === match.params.id).map((product, index) => {
                return (
                    <React.Fragment key={index} >
                        <div className="grid-x grid-padding-x small-up-1 medium-up-2 large-up-2">
                            <ProductImages product={product} />
                            <ProductDetails product={product} />
                        </div>
                        <SimilarProductRenderer />
                    </React.Fragment>)
            }) : <img className="loading-indicator" src={loadingIndicator}></img>}
        </React.Fragment>
    )
}



