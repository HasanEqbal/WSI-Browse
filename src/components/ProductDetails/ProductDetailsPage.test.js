import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import ProductDetailsPage from './ProductDetailsPage';
import ProductImages from './ProductImages';
import ProductDeatils from './ProductDetails';
import SimilarProduct from '../SimilarProducts/SimilarProducts'

const mockStore = configureMockStore();
const store = mockStore({});

const fixture = require('../../testData/testFixture.json');

describe('Component : ProductDetailsPage', () => {

  it('expect to render ProductDetailsPage component', () => {
    const product = fixture.groups[0]
    const match = {
      params: {
        id: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024"
      }
    }
    expect(shallow(<ProductDetailsPage productsList={product} match={match} />)).toMatchSnapshot();
  })

  it('verifying rendering of Product images component', () => {
    const product = fixture.groups[0]
    const match = {
      params: {
        id: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024"
      }
    }
    const wrapper = mount(<ProductDetailsPage productsList={product} match={match} />)
    expect(wrapper.contains(<ProductImages product={product[0]} />)).toBe(true);
  })
  it('verifying rendering of Product details component', () => {
    const product = fixture.groups[0]
    const match = {
      params: {
        id: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024"
      }
    }
    const wrapper = mount(<Provider store={store}><ProductDetailsPage productsList={product} match={match} /></Provider>)
    expect(wrapper.contains(<ProductDeatils product={fixture.groups[0]} />)).toBe(true);
  })
  it('verifying rendering of Similar Product component', () => {
    const product = fixture.groups[0]
    const match = {
      params: {
        id: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024"
      }
    }
    const wrapper = mount(<Provider store={store}><ProductDetailsPage productsList={product} match={match} /></Provider>)
    expect(wrapper.contains(<SimilarProduct />)).toBe(true);
  })

  it('verifying image loader display when data is loading', () => {
    const productlist = {}
    const wrapper = mount(<ProductDetailsPage productsList={productlist} />)
    expect(wrapper.find('img.loading-indicator').length).toBe(1);
  })
})
