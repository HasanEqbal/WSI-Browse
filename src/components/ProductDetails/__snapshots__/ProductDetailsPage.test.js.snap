// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : ProductDetailsPage expect to render ProductDetailsPage component 1`] = `ShallowWrapper {}`;
